docs/source/schemas/streams.rst

```diff
@@ -10,32 +10,61 @@ Overview
 When storing large objects in an AtlasDB database, the underlying
 database can impose limits on the size of the object and its performance
 characteristics. Postgres has a 1GB entry limit, and Cassandra generally
-does not like transmissions of over 30MB over the wire at a time. If it
-is known beforehand that a particular column of a table will store large
-objects, the object and an associated id can be instead stored in a
-stream store table. AtlasDB will automatically segment large objects
-into smaller chunks and store them in a separate stream store table. The
-stored object can then be streamed or updated by the associated id.
-AtlasDB will also reference count these objects and clean them up when
-they are not used anymore. Also, the storage of the streams happens on
-separate transactions and does not count toward your byte limit for your
-local trasaction.
+does not like transmissions of over 30MB over the wire at a time. 
 
-.. code:: java
+If it is known beforehand that a particular column of a table will store
+large objects, the objects should instead be stored in a stream store table.
+AtlasDB will automatically segment large objects in a stream store into 
+smaller chunks, and storage of streams happens on separate transactions and
+does not count toward your byte limit for your local trasaction.
+
+The minimal parameters required to define a stream store are shown below:
 
-    public void addStreamStoreDefinition(String long_name, String short_name, ValueType streamIdType, int inMemoryThreshold);
+.. code:: java
 
-The long name is the name of the stream store in the generated java
-code, the short name the prefix which will be used in the actual
-database tables. The stream id type argument specifies the ValueType of
-the id for each stored object. The in-memory threshold argument
-specifies the largest size object (in bytes) which AtlasDB will cache in
-memory in order to boost retrieval performance.
+    public void addStreamStoreDefinition(
+        new StreamStoreDefinitionBuilder(String shortName, String longName, ValueType valueType).build()
+    )
+    
+- ``shortName`` is the prefix which will be used in the actual database tables.
+- ``longName`` is the name of the stream store in the generated Java code.
+- ``valueType`` specifies the value type of the id for each stored object.
+
+Additional options for the builder include:
+
+- ``hashFirstRowComponent``
```

Comment:
Also, we should note what these first three mean:
* `hashFirstRowComponent`: hashes the "id" component of the table
* `isAppendHeavyAndReadLight`: `false` by default. If set to `true`, some optimisations will be made in Cassandra KVS assuming that these streams will be changed often but not read often.
* `compressBlocksInDb` ... this is never actually used, AFAICT. Let's remove this reference from the docs.